sklearn/preprocessing/data.py

```diff
@@ -2210,6 +2213,17 @@ def __init__(self, n_quantiles=1000, output_distribution='uniform',
         self.random_state = random_state
         self.copy = copy
 
+    @staticmethod
+    def _nanpercentile_force_finite(column_data, percentiles):
+        """Force the output of nanpercentile to be finite."""
+        percentile = nanpercentile(column_data, percentiles)
+        with np.errstate(invalid='ignore'):  # hide NaN comparison warnings
+            if np.all(np.isclose(percentile, np.nan, equal_nan=True)):
```

Comment:
Why not:
```py
if np.all(np.isnan(percentile))
```

If you use that I think you can remove the `with np.errstate(...)`

Reply:
True. no idea how I come to something so complex.